### ISSUE TITLE
Process crashing randomly

### ISSUE THREAD

**Issue by clain23** [OctopusDeploy/Issues#1765]
```
Faulting application name: Octopus.Server.exe, version: 3.0.4.2105, time stamp: 0x55b07f41
Faulting module name: KERNELBASE.dll, version: 6.3.9600.17415, time stamp: 0x54505737
Exception code: 0xe0434352
Fault offset: 0x0000000000008b9c
Faulting process id: 0x226c
Faulting application start time: 0x01d0c5e65f2e922a
Faulting application path: C:\Program Files\Octopus Deploy\Octopus\Octopus.Server.exe
Faulting module path: C:\Windows\system32\KERNELBASE.dll
Report Id: bf7517d2-31d9-11e5-80b7-00155d08d604
Faulting package full name: 
Faulting package-relative application ID: 
```

```
Application: Octopus.Server.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.InvalidOperationException
Stack:
   at Microsoft.WindowsAzure.Common.TransientFaultHandling.RetryPolicy.ExecuteAction[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]](System.Func`1<System.__Canon>)
   at Octopus.Core.RelationalStorage.RelationalStore.BeginTransaction(System.Data.IsolationLevel)
   at Octopus.Server.Orchestration.RelationalTaskRunner.CheckForCancellationOrPause()
   at Octopus.Server.Orchestration.RelationalTaskRunner.RunCancellationCheck()
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object)
   at System.Threading.ThreadHelper.ThreadStart()

```

```
2015-07-24 10:58:20.2892     19 FATAL  Unhandled AppDomain exception occurred: Timeout expired.  The timeout period elapsed prior to obtaining a connection from the pool.  This may have occurred because all pooled connections were in use and max pool size was reached.
System.InvalidOperationException: Timeout expired.  The timeout period elapsed prior to obtaining a connection from the pool.  This may have occurred because all pooled connections were in use and max pool size was reached.
   at System.Data.ProviderBase.DbConnectionFactory.TryGetConnection(DbConnection owningConnection, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal oldConnection, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionInternal.TryOpenConnectionInternal(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
   at System.Data.SqlClient.SqlConnection.TryOpenInner(TaskCompletionSource`1 retry)
   at System.Data.SqlClient.SqlConnection.TryOpen(TaskCompletionSource`1 retry)
   at System.Data.SqlClient.SqlConnection.Open()
   at Microsoft.WindowsAzure.Common.TransientFaultHandling.RetryPolicy.<>c__DisplayClass1.<ExecuteAction>b__0()
   at Microsoft.WindowsAzure.Common.TransientFaultHandling.RetryPolicy.ExecuteAction[TResult](Func`1 func)
   at Octopus.Core.RelationalStorage.RelationalStore.BeginTransaction(IsolationLevel isolationLevel) in y:\work\refs\heads\master\source\Octopus.Core\RelationalStorage\RelationalStore.cs:line 60
   at Octopus.Server.Orchestration.RelationalTaskRunner.CheckForCancellationOrPause() in y:\work\refs\heads\master\source\Octopus.Server\Orchestration\RelationalTaskRunner.cs:line 146
   at Octopus.Server.Orchestration.RelationalTaskRunner.RunCancellationCheck() in y:\work\refs\heads\master\source\Octopus.Server\Orchestration\RelationalTaskRunner.cs:line 136
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart()
```

**Comment by clain23** [comment]
We do not use any Azure things at all.

**Comment by clain23** [comment]
Log file https://www.dropbox.com/s/9dd5l6t6quym5dx/OctopusServer_24_07_2015.txt?dl=0

**Comment by PaulStovell** [comment]
Thanks, this was fixed as part of #1771 and is fixed in 3.0.6
